src/main/java/Orca.java

```diff
@@ -0,0 +1,166 @@
+            commandType = CommandType.EVENT;
+        } else {
+            commandType = CommandType.UNKNOWN;
+        }
+    }
+
+    public static boolean isInputAvailable() {
+        return in.hasNextLine();
+    }
+
+    public static void readUserInput() {
+        userInput = in.nextLine();
+    }
+
+    public static void markTask(int taskNo) {
+        tasks[taskNo - 1].setDone(true);
+        System.out.println("    --------------------------------------------------");
+        System.out.println("    Nice! I've marked this task as done:");
+        System.out.println("      " + tasks[taskNo - 1]);
+        System.out.println("    --------------------------------------------------\n");
+    }
+
+    public static void unmarkTask(int taskNo) {
+        tasks[taskNo - 1].setDone(false);
+        System.out.println("    --------------------------------------------------");
+        System.out.println("    I've marked this task as not done yet:");
+        System.out.println("      " + tasks[taskNo - 1]);
+        System.out.println("    --------------------------------------------------\n");
+    }
+
+    public static void addTask(Task newTask) {
+        tasks[taskCount] = newTask;
+        taskCount++;
+    }
+
+    public static int parseTaskNo(String userInput, int startIdx) {
+        return Integer.parseInt(userInput, startIdx);
+    }
+
+    public static void executeCommand() {
+        switch (commandType) {
+            case BYE:
+                printByeMessage();
+                return;
+            case LIST:
+                printTasks();
+                break;
+            case MARK:
+                taskNo = parseTaskNo(userInput, 5);
+                markTask(taskNo);
+                break;
+            case UNMARK:
+                taskNo = parseTaskNo(userInput, 7);
+                unmarkTask(taskNo);
+                break;
+            case TODO:
+                newTask = new Todo(userInput, 5);
+                addTask(newTask);
+                printLatestAddedTask();
+                break;
+            case DEADLINE:
+                newTask = new Deadline(userInput, 9);
+                addTask(newTask);
+                printLatestAddedTask();
+                break;
+            case EVENT:
+                newTask = new Event(userInput, 6);
+                addTask(newTask);
+                printLatestAddedTask();
+                break;
+            default:
+                printUnknownCommandMessage();
+                break;
+        }
+    }
+
+    public static void runOrca() {
+        while (isInputAvailable()) {
+            readUserInput();
+            findCommandType();
+            executeCommand();
+        }
+    }
+
+    public static void finishProcess() {
+        in.close();
+    }
+
+    public static void main(String[] args) {
+        printGreetingMessage();
+        runOrca();
+        printByeMessage();
+        finishProcess();
+    }
```

Comment:
Perhaps you could add some comments to clarify how your code works?

Reply:
I will add some comments. Thank you.